install-deps.sh

```diff
@@ -146,19 +146,48 @@ function install_pkg_on_ubuntu {
     fi
 }
 
+boost_ver=1.82
+
+function clean_boost_on_ubuntu {
+    ci_debug "Running clean_boost_on_ubuntu() in install-deps.sh"
+    # Find currently installed version. This may not work if there are
+    # multiple versions installed.
+    local installed_ver=$(apt -qq list --installed ceph-libboost*-dev 2>/dev/null |
+                              grep -e 'libboost[0-9].[0-9]\+-dev' |
+                              cut -d' ' -f2 |
+                              cut -d'.' -f1,2)
+    # If installed_ver contains a space, we can't really count on it,
+    # but otherwise, bail out if the version installed is the version
+    # we want.
+    if test -n "$installed_ver" && echo "$installed_ver" | grep -v -q ' '; then
```

Comment:
Are you sure that `installed_ver` would actually contain spaces?  I just tried a similar command and it's newlines instead of spaces.  This matters since the variable expansion is quoted here:
```
$ foo=$(apt -qq list --installed emacs* 2>/dev/null | cut -d' ' -f2 | cut -d'.' -f1,2)
$ echo $foo
1:26.3+1-1ubuntu2 1:26.3+1-1ubuntu2 1:26.3+1-1ubuntu2 1:26.3+1-1ubuntu2 3.0
$ echo "$foo"
1:26.3+1-1ubuntu2
1:26.3+1-1ubuntu2
1:26.3+1-1ubuntu2
1:26.3+1-1ubuntu2
3.0
```

Reply:
And further, since there would be multiple packages with the same version (atomic, chrono, container, etc), wouldn't this condition always trigger (assuming the separator that it's looking for is correct)?

Reply:
No. The grep filters them out.

Reply:
You are correct about the space. Switching to `[[:space:]]`

Reply:
The grep does not actually filter what we'd like, though, so I'll use uniq.